SmartTutor/KnowledgeComponentModel/KnowledgeComponents/KnowledgeComponentProgress.cs

```diff
@@ -0,0 +1,17 @@
+﻿using SmartTutor.LearnerModel.Learners;
+
+namespace SmartTutor.KnowledgeComponentModel.KnowledgeComponents
+{
+    public class KnowledgeComponentProgress
+    {
+        public int Id { get; private set; }
+        
+        public double Progress { get; private set; }
```

Comment:
We should keep this field in mind as we _progress_ through our R&D :)

Namely, this might better be suited as a [value object](https://en.wikipedia.org/wiki/Value_object) that constrains the possible values from 0 to 1 (or from 0 to 100). Currently this implementation has no such limitation, and we can have -3566 as easily as 0.5.